From ff416cce47e29c113c706684a05ebb112c9116b3 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Tue, 7 Apr 2009 04:18:32 +0200
Subject: [PATCH] Make Terminate_Server on ctrl+alt+bksp an option

This disables zapping by default, it can be reenabled with
'setxkbmap -option terminate:ctrl_alt_bksp'.
---
 rules/base.o_s.part |    1 +
 rules/base.xml.in   |   12 ++++++++++++
 symbols/pc          |    5 +----
 symbols/srvr_ctrl   |   13 +++++++++----
 4 files changed, 23 insertions(+), 8 deletions(-)

diff --git a/rules/base.o_s.part b/rules/base.o_s.part
index c401790..9843c61 100644
--- a/rules/base.o_s.part
+++ b/rules/base.o_s.part
@@ -97,4 +97,5 @@
   shift:breaks_caps     =	+shift(breaks_caps)
   esperanto:qwerty      =	+epo(qwerty)
   esperanto:dvorak      =	+epo(dvorak)
+  terminate:ctrl_alt_bksp =	+srvr_ctrl(terminate_ctrl_alt_bksp)
 
diff --git a/rules/base.xml.in b/rules/base.xml.in
index a7616c5..fe7a26f 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -4914,5 +4914,17 @@
        </configItem>
       </option>
     </group>
+    <group allowMultipleSelection="true">
+      <configItem>
+        <name>terminate</name>
+        <_description>Key sequence to kill the X server</_description>
+      </configItem>
+      <option>
+        <configItem>
+          <name>terminate:ctrl_alt_bksp</name>
+          <_description>Control + Alt + Backspace</_description>
+        </configItem>
+      </option>
+    </group>
   </optionList>
 </xkbConfigRegistry>
diff --git a/symbols/pc b/symbols/pc
index 07472b8..9474df5 100644
--- a/symbols/pc
+++ b/symbols/pc
@@ -19,10 +19,7 @@ xkb_symbols "pc105" {
     include "pc(editing)"
     include "keypad(x11)"
 
-    key <BKSP> {
-        type="CTRL+ALT",
-        symbols[Group1]= [ BackSpace,   Terminate_Server ]
-    };
+    key <BKSP> {	[ BackSpace 	]	};
 
     key  <TAB> {	[ Tab,	ISO_Left_Tab	]	};
     key <RTRN> {	[ Return	]	};
diff --git a/symbols/srvr_ctrl b/symbols/srvr_ctrl
index 712f763..9b333e3 100644
--- a/symbols/srvr_ctrl
+++ b/symbols/srvr_ctrl
@@ -13,10 +13,7 @@ xkb_symbols "xfree86" {
 partial keypad_keys 
 xkb_symbols "stdkeypad" {
 
-    key <BKSP> {
-	type="CTRL+ALT",
-	symbols[Group1]= [ BackSpace,   Terminate_Server ]
-    };
+    include "srvr_ctrl(terminate_ctrl_alt_bksp)"
 
 // Ungrab cancels server/keyboard/pointer grabs
     key <KPDV> {
@@ -44,6 +41,14 @@ xkb_symbols "stdkeypad" {
 
 };
 
+partial modifier_keys
+xkb_symbols "terminate_ctrl_alt_bksp" {
+    key <BKSP> {
+	type="CTRL+ALT",
+	symbols[Group1]= [ NoSymbol,   Terminate_Server ]
+    };
+};
+
 partial function_keys 
 xkb_symbols "fkey2vt" {
 
-- 
1.5.6.3

